RecoLocalCalo/HcalRecProducers/plugins/alpaka/Mahi.dev.cc

```diff
@@ -0,0 +1,1457 @@
+                                      HcalRecoParamWithPulseShapeDevice::ConstView recoParamsWithPS,
+                                      bool const useEffectivePedestals,
+                                      int const sipmQTSShift,
+                                      int const sipmQNTStoSum,
+                                      int const firstSampleShift,
+                                      float const ts4Thresh,
+                                      float* amplitudes,
+                                      float* noiseTerms,
+                                      float* electronicNoiseTerms,
+                                      int8_t* soiSamples,
+                                      int const windowSize) const {
+          auto const nchannelsf015 = f01HEDigis.size() + f5HBDigis.size();
+          auto const startingSample = compute_nsamples<Flavor1>(f01HEDigis.stride()) - windowSize;
+          auto const nchannels = f01HEDigis.size() + f5HBDigis.size() + f3HBDigis.size();
+
+          //first index = groups of channel
+          auto const nchannels_per_block(alpaka::getWorkDiv<alpaka::Block, alpaka::Elems>(acc)[0u]);
+          //2nd index = groups of sample
+          auto const nsamplesForCompute(alpaka::getWorkDiv<alpaka::Block, alpaka::Elems>(acc)[1u]);
+
+          // configure shared mem
+          float* shrEnergyM0PerTS = alpaka::getDynSharedMem<float>(acc);
+          float* shrChargeMinusPedestal = shrEnergyM0PerTS + nsamplesForCompute * nchannels_per_block;
+          float* shrMethod0EnergyAccum = shrChargeMinusPedestal + nsamplesForCompute * nchannels_per_block;
+          float* shrEnergyM0TotalAccum = shrMethod0EnergyAccum + nchannels_per_block;
+          unsigned long long int* shrMethod0EnergySamplePair =
+              reinterpret_cast<unsigned long long int*>(shrEnergyM0TotalAccum + nchannels_per_block);
+
+          //Loop over all groups of channels
+          for (auto group : uniform_groups_y(acc, nchannels)) {
+            //Loop over each channel, first compute soiSamples and shrMem for all channels
+            for (auto channel : uniform_group_elements_y(acc, group, nchannels)) {
+              auto const gch = channel.global;
+              auto const lch = channel.local;
+
+              for (auto i_sample : independent_group_elements_x(acc, nsamplesForCompute)) {
+                auto const sampleWithinWindow = i_sample;
+                auto const sample = i_sample + startingSample;
+                auto const linearThPerBlock = i_sample + channel.local * nsamplesForCompute;
+                // initialize
+                if (sampleWithinWindow == 0) {
+                  soiSamples[gch] = -1;
+                  shrMethod0EnergyAccum[lch] = 0;
+                  //TODO: check this conversion
+                  shrMethod0EnergySamplePair[lch] = float_as_uint(std::numeric_limits<float>::min());
```

Comment:
I think this can simply be
```suggestion
                  shrMethod0EnergySamplePair[lch] = 0;
```

As far as I understand it is only used to set the starting value, and any value larger than 0 (or 1.17549e-38 originally) will then replace it.